Generate a series of 100,000 floats in a normal distribution with a mean of 0 and a standard deviation of 100.

Get the descriptive statistics for this series. How close are the mean and median? (You don’t need to calculate the difference; rather, consider why they aren’t the same.)

Replace the minimum value with 5 times the maximum value.

Get the descriptive statistics again. Did the mean, median, and standard deviations change from their previous values? (Again, it’s enough to see the difference without calculating it.) If so, why?

In [1]:
import numpy as np
import pandas as pd

In [2]:
our_series = pd.Series(np.random.normal(0, 100, 100000))
our_series

0        -44.916742
1        -91.888363
2        -27.225054
3         -4.128708
4         25.986617
            ...    
99995     50.496350
99996    -76.775758
99997   -178.534603
99998   -264.699106
99999    -73.932361
Length: 100000, dtype: float64

Get the descriptive statistics for this series. How close are the mean and median?

In [3]:
our_series.describe()

count    100000.000000
mean          0.239918
std          99.730967
min        -478.392597
25%         -66.771073
50%           0.398717
75%          67.489288
max         442.545639
dtype: float64

In [4]:
our_series.median()

0.3987167148011155

Replace the minimum value with 5 times the maximum value.

In [5]:
our_series[our_series == our_series.min()] = our_series.max() * 5

Get the descriptive statistics again. Did the mean, median, and standard deviations change from their previous values?

In [6]:
our_series.describe()

count    100000.000000
mean          0.266829
std          99.964622
min        -434.350136
25%         -66.767046
50%           0.401064
75%          67.492611
max        2212.728194
dtype: float64

In [7]:
our_series.median()

0.4010643089869247

The median barely shifted. That’s because it tends to be the most stable measurement, even when we have fluctuations at the extremes.

Demonstrate that 68%, 95%, and 99.7% of the values in s are indeed within one, two, and three standard distributions of the mean.

In [8]:
# python
mean = our_series.mean()
std = our_series.std()

for k in (1, 2, 3):
    print(((our_series - mean).abs() <= k * std).mean() * 100)

68.49
95.491
99.703


Calculate the mean of numbers greater than s.mean(). Then calculate the mean of numbers less than s.mean(). Is the average of these two numbers the same as s.mean()?

In [32]:
a_mean = our_series[our_series < our_series.mean()].mean()
c_mean = our_series[our_series > our_series.mean()].mean()
b_mean = our_series.mean()

(a_mean + c_mean)/2 == b_mean

np.False_

What is the mean of the numbers beyond three standard deviations?

In [34]:
our_series[our_series > 3*our_series.std()].mean()

np.float64(343.253029591812)